graylog2-server/src/main/java/org/graylog2/rest/models/messages/responses/DecorationStats.java

```diff
@@ -53,9 +53,10 @@
     @JsonProperty(FIELD_CHANGED_FIELDS)
     public Map<String, Object> changedFields() {
         return Sets.intersection(originalMessage().keySet(), decoratedMessage().keySet())
-            .stream()
-            .filter(key -> !originalMessage().get(key).equals(decoratedMessage().get(key)))
-            .collect(Collectors.toMap(Function.identity(), key -> originalMessage().get(key)));
+                .stream()
+                .filter(key -> !originalMessage().get(key).equals(decoratedMessage().get(key)))
```

Comment:
This method's implementation, as well as the **addedFields**() and **removedFields**() seem out of place in a domain object.  Would these values be best supplied by the builder or an external utility class.

Further, could these methods benefit from unit tests?

Reply:
If this object was larger and had other things inside of it I'd think the use of a builder or external utility class would make sense. Since those 3 methods are really the only use of this class and they aren't all that verbose, I don't see an issue with them left as is. Doesn't matter to me one way or the other if there happens to be a consensus on refactoring.

I can add some unit tests to validate they are created correctly though.